### PR TITLE
create global layout (header, footer, and more)

### PR DESCRIPTION
**Pull request by braxex**
## Description

Create global layout, including header and footer, and adds appropriate unit and integration tests. It looks like a ton of lines, but that's like 80% `yarn.lock` fun, so don't be too alarmed.

Incidentally introduces the following:
- New reusable custom components (Menu, MenuOption) and custom wrappers (RouterLink, Icon).
- Tabbed navigation focus functionality.
- New custom hooks (useBreakpoint, makeMediaQueryClasses) to replace media queries and simplify responsive styling efforts.
- Global styling baseline and patterns for styling solution.
- Theme breakpoints.
- Integration testing functionality and sets baseline.

Satisfies the following tickets: 
- [MDCT-168: MCR Global Header](https://qmacbis.atlassian.net/browse/MDCT-168)
- [MDCT-169: MCR Global Footer](https://qmacbis.atlassian.net/browse/MDCT-169)

### How to test

Prerequisite: Ensure correct `.env` (see below).

To test UI:
1. `cd` && `nvm use`
2. `./dev local`
3. Check things out and try to break it (mess with responsive breakpoints, click around, etc.). Note: All routes outside of `/` are non-functional.

Unit tests:
1. `cd services/ui-src && nvm use`
2. `yarn test`
3. All tests should be passing

Integration tests:
1. `cd` && `nvm use`
2. `yarn test`
3. When cypress window launches, click on any test suite to run.
4. All tests should be passing and there should be no accessibility errors or warnings.

### Changed Dependencies

Just a little `yarn upgrade` action for `/`, `/services/ui-src`, and `/tests/cypress`.

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research) tests
- [x] ~~I have added analytics, if necessary~~
- [x] ~~I have updated the documentation, if necessary~~

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [x] I have done the lgtm review

***
***
#### root `.env`
```
SKIP_PREFLIGHT_CHECK=true
LOCAL_LOGIN=true
MEASURE_TABLE_NAME=local-measures
MEASURE_TABLE_ARN=local_nonsense_if_unset_we_search_CF_for
measureTableName=local-measures
DYNAMODB_URL=http://localhost:8000/
API_URL=http://localhost:3030/local
S3_LOCAL_ENDPOINT=http://localhost:4569/
S3_ATTACHMENTS_BUCKET_NAME=local-uploads
URL=http://localhost/3000
```

#### services/ui-src `.env`
```
LOCAL_LOGIN=true
API_REGION=us-east-1
API_URL=http://localhost:3030/
COGNITO_REGION=us-east-1
COGNITO_IDENTITY_POOL_ID=us-east-1:61c849d8-f95c-45af-8131-15b75c525034
COGNITO_USER_POOL_ID=us-east-1_7U4ihGYZP
COGNITO_USER_POOL_CLIENT_ID=46fh6sftchq768ab0v65cpq0cf
COGNITO_USER_POOL_CLIENT_DOMAIN=master-login-46fh6sftchq768ab0v65cpq0cf.auth.us-east-1.amazoncognito.com
COGNITO_REDIRECT_SIGNIN=http://localhost:3000/
COGNITO_REDIRECT_SIGNOUT=http://localhost:3000/
S3_ATTACHMENTS_BUCKET_REGION=us-east-1
S3_ATTACHMENTS_BUCKET_NAME=uploads-master-attachments-024259748323
S3_LOCAL_ENDPOINT=http://localhost:4569/
```

### CODE REVIEW

**Review by karla-vm — APPROVED** [review]
Everything is looking clean! And all the tests pass. Great job!!! :D

File: services/ui-src/src/index.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import ReactDOM from "react-dom";
 import { BrowserRouter as Router } from "react-router-dom";
-import { ReactQueryDevtools } from "react-query/devtools";
+// import { ReactQueryDevtools } from "react-query/devtools";
```
Comment (TheDanMiller) [inline services/ui-src/src/index.tsx:3]: Did you mean to check in the commented code

File: services/ui-src/src/components/layout/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import {
+  makeMediaQueryClasses,
+  useBreakpoint,
+} from "../../utils/useBreakpoint";
+// assets
+import hhsLogo from "../../assets/logo_hhs.svg";
+import medicaidLogo from "../../assets/logo_medicaid.svg";
+
+export const Footer = () => {
+  const { isMobile, isDesktop } = useBreakpoint();
```
[the remark this answers is not in the document]
Comment (braxex) [reply services/ui-src/src/components/layout/Footer.tsx:14]: Yep both needed. `isMobile` will be true if viewport is in mobile range (<=560px) and `isDesktop` will be true if viewport is in desktop range (>880px). So you can target mobile and tablet together with `!isDesktop`. Alternatively here we could have imported `isMobile` and `isTablet` (true if viewport is in tablet range (>580px & <=880px) and used `isMobile || isTablet` but I thought `!isDesktop` was slightly cleaner. 

I also created a `isUltrawide` (true if >1600px) so `isDesktop` and `isUltrawide` can be true at the same time, but those are the only two that overlap. 👍

File: services/ui-src/src/components/layout/Footer.test.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  test("FAQ link is visible", () => {
+    expect(screen.getByTestId("faq-link")).toBeVisible();
+  });
+
+  test("Accessibility statement link is visible", () => {
+    expect(screen.getByTestId("accessibility-statement-link")).toBeVisible();
+  });
+});
+
+describe("Test Footer accessibility", () => {
```
[the remark this answers is not in the document]
Comment (braxex) [reply services/ui-src/src/components/layout/Footer.test.tsx:32]: Cool. Let's chat whenever.

File: services/ui-src/src/index.tsx (hunk 1 above)
Comment (braxex) [reply services/ui-src/src/index.tsx:3]: Haha yep I did. I'm guessing the ReactQuery dev tools will be nice to use down the road in local dev, but if it's in the DOM like this it drops an icon in the bottom left that prevents some of our E2E testing from working locally. I suppose we could use a local env variable to flip it off if running E2E tests though.

**Pull request merged**